### PR TITLE
NAS-114326 / Bluefin / Refactor Jobs related components

### PR DESCRIPTION
**Pull request by denysbutenko**
Refactor JobsManager and JobsList to consume data from the global store.

- `JobsManagerStore` and `JobsListStore` are removed
- Use `Task` in text for users and `Job` internally for components

For testing:

- Check `Task Manager` for issues/regressions
- Click on the `History` button and check the `Tasks` page for issues/regressions

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
Looks nice. Please confirm wording changes with Damian.

**Comment by denysbutenko** [comment]
@dszidi in master/bluefin on the `Jobs Panel` the title says `Task Manager`, but it refers to `Jobs` in other places. I changed the title from `Jobs` to `Tasks` on `Jobs List` page. Does it make sense?

**Comment by denysbutenko** [comment]
squashed & rebased

**Review by dszidi — COMMENTED** [review]
Seems to get into this weird state when I hit the View Arguments button followed by the View Logs button

image

**Comment by denysbutenko** [comment]
> Seems to get into this weird state where the row height expands to fill the element height. Happens when I hit the View Arguments button followed by the View Logs button
> 
> image

Thanks, fixed!

**Comment by denysbutenko** [comment]
I will fix unit tests but have had no success yet. `jasmine is not defined` if you are familiar with such cases, let me know.

File: src/app/modules/jobs/components/jobs-panel/jobs-panel.component.spec.ts [hunk 1]
```diff
@@ -1,76 +1,89 @@
 import { HarnessLoader } from '@angular/cdk/testing';
 import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
 import { MatButtonHarness } from '@angular/material/button/testing';
-import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MatDialogRef } from '@angular/material/dialog';
 import { Router } from '@angular/router';
-import {
-  createRoutingFactory, mockProvider, Spectator,
-} from '@ngneat/spectator/jest';
+import { Spectator, createRoutingFactory, mockProvider } from '@ngneat/spectator';
```
Comment (undsoft) [inline src/app/modules/jobs/components/jobs-panel/jobs-panel.component.spec.ts:6]: Change to '@ngneat/spectator/jest'

Comment (denysbutenko) [reply src/app/modules/jobs/components/jobs-panel/jobs-panel.component.spec.ts]: I stuck for a day with this. Thank you 🙏🏻

**Review by undsoft — CHANGES_REQUESTED** [review]
1. Layout is a bit weird.
2. I am on the fence about renaming to tasks. We use jobs everywhere else in the system and we already use 'tasks' for things like replication tasks.
https://user-images.githubusercontent.com/1481270/152552240-65d293be-1e06-4c18-b7c0-71b73cb25930.mov

**Comment by denysbutenko** [comment]
> 1. Layout is a bit weird.
> 2. I am on the fence about renaming to tasks. We use jobs everywhere else in the system and we already use 'tasks' for things like replication tasks.
>    https://user-images.githubusercontent.com/1481270/152552240-65d293be-1e06-4c18-b7c0-71b73cb25930.mov

1. Good catch, thanks
2. I'm okay to keep it as is `Task Manager` on `Jobs Panel` and `Jobs` on `Jobs List` when clicking on the `History` button.

**Comment by undsoft** [comment]
Why not use the same terminology?

**Comment by denysbutenko** [comment]
That's what I wanted initially :)

Long story short:
1. The panel before refactoring was titled `Task Manager`
2. I did refactoring and we've changed the title to `Running Jobs` [PR](https://github.com/truenas/webui/pull/5656)
3. At some point, we get a request to add `Waiting` jobs to that panel, and the title was changed to `Task Manager` and it is currently in the master/bluefin branch. [Change](https://github.com/truenas/webui/commit/c1b294fc6f0ae4a09a440dcd5d19516e5320f5ff#diff-6cad58c6ce46ed6418a64353a254a3f0d8c03bccf70c52a42ce1ac32bd7710bdL3)
4. That's why I am supposed to use Tasks is more suitable for users. In this PR, I wanted to sync user strings for Jobs Panel and Jobs List to use `Task Manager` and `Tasks`.
5. Seems like the appropriate way to achieve consistency is to change `Task Manager` to `Jobs`?

**Comment by undsoft** [comment]
I like the idea of just calling it jobs for consistency. @dszidi ?

**Comment by denysbutenko** [comment]
image

**Review by undsoft — APPROVED** [review]
Looks good to me.

**Comment by dszidi** [comment]
> I like the idea of just calling it jobs for consistency. @dszidi ?



Works for me.

**Pull request merged**